plugins/http/src/commands.rs

```diff
@@ -20,16 +20,122 @@ pub struct FetchResponse {
     url: String,
 }
 
-#[command]
-pub async fn fetch<R: Runtime>(
-    app: AppHandle<R>,
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ClientConfig {
     method: String,
     url: url::Url,
     headers: Vec<(String, String)>,
     data: Option<Vec<u8>>,
     connect_timeout: Option<u64>,
     max_redirections: Option<usize>,
+    proxy: Option<ProxyConfig>,
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum ProxyConfig {
+    All {
+        all: UrlOrConfig,
+    },
+    HttpAndHttps {
+        http: UrlOrConfig,
+        https: UrlOrConfig,
+    },
+    Http {
+        http: UrlOrConfig,
+    },
+    Https {
+        https: UrlOrConfig,
+    },
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[serde(untagged)]
+pub enum UrlOrConfig {
+    Url(String),
+    Config(DetailedProxyConfig),
+}
+
+#[derive(Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DetailedProxyConfig {
```

Comment:
Let's rename this to `ProxyConfig` 
```suggestion
pub struct ProxyConfig {
```
and rename `ProxyConfig` to just `Proxy` wich will conflict with `Proxy` from `reqwest` so just use the qualified path for that, i.e. `reqwest::Proxy`